src/coreclr/System.Private.CoreLib/src/System/MulticastDelegate.cs

```diff
@@ -436,15 +436,15 @@ public sealed override Delegate[] GetInvocationList()
         internal new bool HasSingleTarget => !(_invocationList is object[]);
 
         // Used by delegate invocation list enumerator
-        internal Delegate? TryGetAt(int index)
+        internal object? /* Delegate? */ TryGetAt(int index)
         {
             if (!(_invocationList is object[] invocationList))
             {
                 return (index == 0) ? this : null;
             }
             else
             {
-                return ((uint)index < (uint)_invocationCount) ? (Delegate)invocationList[index] : null;
+                return ((uint)index < (uint)_invocationCount) ? invocationList[index] : null;
```

Comment:
In what situation does _invocationCount != invocationList.Length?  If they were the same, you could compare against invocationList.Length here instead and avoid a bounds check.

Reply:
The multicast array can have extra elements to amortize the cost of resizing the array for combine and remove operations.